### PR TITLE
nrf5x: update rtc to new regs

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request updates nrf5x rtc to new regs.

### Testing Strategy

This pull request was tested by compiling.

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Kernel: Updated the relevant files in `/docs`, or no updates are required.
- [x] Userland: Added/updated the application README, if needed.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: chips/nrf5x/src/rtc.rs [hunk 1]
```diff
@@ -1,28 +1,102 @@
...
+    /// Interrupt enable set register.
+    intenset: ReadWrite<u32, Inte::Register>,
+    /// Interrupt enable clear register.
+    intenclr: ReadWrite<u32, Inte::Register>,
+    _reserved3: [u8; 52],
+    /// Configures event enable routing to PPI for each RTC event.
+    evten: ReadWrite<u32, Inte::Register>,
+    /// Enable events routing to PPI. The reading of this register gives the value of EV
+    evtenset: ReadWrite<u32, Inte::Register>,
+    /// Disable events routing to PPI. The reading of this register gives the value of E
```
Comment (niklasad1) [inline chips/nrf5x/src/rtc.rs]: I don't understand `The reading of this register gives the value of E`! But I guess the value means which events that are `enabled`

Comment (niklasad1) [reply chips/nrf5x/src/rtc.rs]: Or is this auto-generated?! Can't find it in the svd file!

File: chips/nrf5x/src/rtc.rs [hunk 2]
```diff
@@ -1,28 +1,102 @@
...
+    evtenclr: ReadWrite<u32, Inte::Register>,
+    _reserved4: [u8; 440],
+    /// Current COUNTER value.
+    counter: ReadOnly<u32>,
+    /// 12-bit prescaler for COUNTER frequency (32768/(PRESCALER+1)). Must be written wh
```
Comment (niklasad1) [inline chips/nrf5x/src/rtc.rs]: Parse error I guess should be `12 bit prescaler for COUNTER frequency (32768/(PRESCALER+1)).Must be written when RTC is stopped`

**Comment by bradjc** [comment]
Updated. Clearly the `svd2regs.py` script doesn't work very well for the nrf.

File: chips/nrf5x/src/rtc.rs [hunk 3]
```diff
@@ -1,28 +1,100 @@
...
+
 pub struct Rtc {
-    callback: Cell<Option<&'static time::Client>>,
+    regs: *const RtcRegisters,
+    callback: OptionalCell<&'static time::Client>,
 }
 
 pub static mut RTC: Rtc = Rtc {
-    callback: Cell::new(None),
+    regs: RTC1_BASE as *const RtcRegisters,
```
Comment (alevy) [inline chips/nrf5x/src/rtc.rs]: Actually, these days we should start using `StaticRef`... as long as we're at it

**Pull request merged**